### ISSUE TITLE
inference nan?

### ISSUE THREAD

**Issue by wutheringcoo** [HorizonRobotics/Sparse4D#53]
halo, 我在mini训练集上训练了两个iter[batch=1]，也就是训练了两张图，loss,grad都是正常的无nan出现，于是保存了一个pth，想用这个pth测试下test链路。在mini测试集发现一个现象，推理的前38张图都是没有问题的，测试第39张图时，发现推理链路走到head: operation_order第21个op： [deformabl]e，所有的输入和对应layer的网络参数都是没有nan的，但是推理结果instance_feature会出现nan，这个怎么理解呀？

这个nan会导致后面所有推理结果如下：
```
{'img_bbox': {'boxes_3d': tensor([[nan, nan, nan,  ..., nan, nan, nan],
        [nan, nan, nan,  ..., nan, nan, nan],
        [nan, nan, nan,  ..., nan, nan, nan],
        ...,
        [nan, nan, nan,  ..., nan, nan, nan],
        [nan, nan, nan,  ..., nan, nan, nan],
        [nan, nan, nan,  ..., nan, nan, nan]]), 
'scores_3d': tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,..., nan, nan, nan, nan]),
 'labels_3d': tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,..., 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), 'cls_scores': tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,...
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan]),
 'instance_ids': tensor([12600, 12601, 12602, 12603, 12604, 12605, 12606, 12607, 12608, 12609,
        12610, 12611, 12612, 12613, 12614, 12615, 12616, 12617, 12618, 12619,
        12620, 12621, 12622, 12623, 12624, 12625, 12626, 12627, 12628, 12629,
        12630, 12631, 12632, 12633, 12634, 12635, 12636, 12637, 12638, 12639,
        12640, 12641, 12642, 12643, 12644, 12645, 12646, 12647, 12648, 12649,
        12650, 12651, 12652, 12653, 12654, 12655, 12656, 12657, 12658, 12659,
        12660, 12661, 12662, 12663, 12664, 12665, 12666, 12667, 12668, 12669,
        12670, 12671, 12672, 12673, 12674, 12675, 12676, 12677, 12678, 12679,
        12680, 12681, 12682, 12683, 12684, 12685, 12686, 12687, 12688, 12689,
        12690, 12691, 12692, 12693, 12694, 12695, 12696, 12697, 12698, 12699,
        12700, 12701, 12702, 12703, 12704, 12705, 12706, 12707, 12708, 12709,
        12710, 12711, 12712, 12713, 12714, 12715, 12716, 12717, 12718, 12719,
        12720, 12721, 12722, 12723, 12724, 12725, 12726, 12727, 12728, 12729,
        12730, 12731, 12732, 12733, 12734, 12735, 12736, 12737, 12738, 12739,
        12740, 12741, 12742, 12743, 12744, 12745, 12746, 12747, 12748, 12749,
        12750, 12751, 12752, 12753, 12754, 12755, 12756, 12757, 12758, 12759,
        12760, 12761, 12762, 12763, 12764, 12765, 12766, 12767, 12768, 12769,
        12770, 12771, 12772, 12773, 12774, 12775, 12776, 12777, 12778, 12779,
        12780, 12781, 12782, 12783, 12784, 12785, 12786, 12787, 12788, 12789,
        12790, 12791, 12792, 12793, 12794, 12795, 12796, 12797, 12798, 12799,
        12800, 12801, 12802, 12803, 12804, 12805, 12806, 12807, 12808, 12809,
        12810, 12811, 12812, 12813, 12814, 12815, 12816, 12817, 12818, 12819,
        12820, 12821, 12822, 12823, 12824, 12825, 12826, 12827, 12828, 12829,
        12830, 12831, 12832, 12833, 12834, 12835, 12836, 12837, 12838, 12839,
        12840, 12841, 12842, 12843, 12844, 12845, 12846, 12847, 12848, 12849,
        12850, 12851, 12852, 12853, 12854, 12855, 12856, 12857, 12858, 12859,
        12860, 12861, 12862, 12863, 12864, 12865, 12866, 12867, 12868, 12869,
        12870, 12871, 12872, 12873, 12874, 12875, 12876, 12877, 12878, 12879,
        12880, 12881, 12882, 12883, 12884, 12885, 12886, 12887, 12888, 12889,
        12890, 12891, 12892, 12893, 12894, 12895, 12896, 12897, 12898, 12899],
       device='cuda:0')}},
```

测试的代码及结果如下：
operation_order=[
            'deformable', 'ffn', 'norm', 'refine', 'temp_gnn', 'gnn', 'norm',
            'deformable', 'ffn', 'norm', 'refine', 'temp_gnn', 'gnn', 'norm',
            'deformable', 'ffn', 'norm', 'refine', 'temp_gnn', 'gnn', 'norm',
            'deformable', 'ffn', 'norm', 'refine', 'temp_gnn', 'gnn', 'norm',
            'deformable', 'ffn', 'norm', 'refine', 'temp_gnn', 'gnn', 'norm',
            'deformable', 'ffn', 'norm', 'refine'],

 elif op == "deformable": 
            # when i = 21, layer parameters have no nan
             for name, t in (self.layers[i].state_dict()).items():
                    if (torch.isnan(t)).any():
                        print(name)
                        print(t)
                        print("ERROR")
            instance_feature = self.layers[i](
                instance_feature, 
                anchor, 
                anchor_embed, 
                feature_maps, 
                metas, 
            )

# All  input is below, no one have none
# instance_feature is like this: (torch.isnan(instance_feature)).any() = False
tensor([[[-0.6244,  0.6917,  1.3716,  ..., -0.8951,  0.5399, -1.1545],
         [-0.4455,  1.0037,  1.1536,  ..., -0.6888,  0.4201, -0.6402],
         [-0.6256,  0.9606,  1.3175,  ..., -0.6508,  0.5477, -0.7793],
         ...,
         [-0.5452, -0.2422,  0.0628,  ..., -1.0446,  1.0575,  0.2228],
         [-0.9898,  0.0861,  0.4529,  ...,  0.3991,  1.1602, -0.1907],
         [-0.9038,  0.6720,  1.7239,  ..., -0.5836,  0.6197,  0.5593]]],
       device='cuda:0')

# anchor is like this: (torch.isnan(anchor)).any() = False
tensor([[[  7.5257,  35.5041,  -2.5089,  ...,  -2.6111,  -1.5984,  -9.1952],
         [  5.9434,  24.3498,  -2.9597,  ...,  -1.8038,  -0.2479,  -9.1792],
         [  8.5267,  37.7096,  -2.5505,  ...,  -3.0587,  -1.0735,  -8.8504],
         ...,
         [ 24.9640,  31.1047,  -0.8030,  ...,   0.0523,   0.7762,  -2.2962],
         [-36.7076,  20.1978,   0.3171,  ...,   0.6775,   1.3946,  -0.0863],
         [ -7.7003,  12.8096,  -0.6191,  ...,  -0.1173,   0.8939,  -0.6261]]],
       device='cuda:0')

# anchor_embed is like this: (torch.isnan(anchor_embed)).any() = False
tensor([[[-0.6151,  1.3798,  3.6124,  ...,  0.4734,  1.2472,  0.4988],
         [-0.6144,  1.3919,  3.6019,  ...,  0.6278,  0.6186,  0.8727],
         [-0.6175,  1.3635,  3.5322,  ...,  0.5120,  1.2253,  0.7744],
         ...,
         [-0.7040, -0.6567, -0.7038,  ..., -0.6619, -0.6618,  2.3825],
         [-0.6119,  0.9332, -0.6117,  ..., -0.5855, -0.5854,  3.8435],
         [-0.4049,  0.8270,  1.8842,  ..., -0.6064, -0.6064,  3.7762]]],
       device='cuda:0')

# feature_maps is like this , no one have none
[tensor([[[ 0.3695, -0.1179,  0.3099,  ...,  0.6223, -0.1194,  0.9810],
         [ 0.1660, -0.0323,  0.6601,  ...,  1.3133,  0.2247,  0.8307],
         [ 0.2859, -0.2132,  0.4637,  ...,  1.2302,  0.0369,  0.9049],
         ...,
         [ 0.6084, -0.4893, -0.2446,  ...,  0.4528, -0.0345, -0.4409],
         [ 0.6983, -0.2620, -0.0791,  ...,  0.3225, -0.2435, -0.7912],
         [ 0.6109,  0.3830,  0.3729,  ...,  0.0256, -0.3553, -1.0339]]],
       device='cuda:0'), tensor([[[ 64, 176],
         [ 32,  88],
         [ 16,  44],
         [  8,  22]],

        [[ 64, 176],
         [ 32,  88],
         [ 16,  44],
         [  8,  22]],

        [[ 64, 176],
         [ 32,  88],
         [ 16,  44],
         [  8,  22]],

        [[ 64, 176],
         [ 32,  88],
         [ 16,  44],
         [  8,  22]],

        [[ 64, 176],
         [ 32,  88],
         [ 16,  44],
         [  8,  22]],

        [[ 64, 176],
         [ 32,  88],
         [ 16,  44],
         [  8,  22]]], device='cuda:0'), tensor([[    0, 11264, 14080, 14784],
        [14960, 26224, 29040, 29744],
        [29920, 41184, 44000, 44704],
        [44880, 56144, 58960, 59664],
        [59840, 71104, 73920, 74624],
        [74800, 86064, 88880, 89584]], device='cuda:0')]

**Comment by linxuewu** [comment]
emmmm，这个原因我也不太好判断...为什么要做这样的测试呢？
